File: spring_java/src/main/java/dev/pollito/spring_java/sakila/film/domain/port/in/FilmUseCases.java
package dev.pollito.spring_java.sakila.film.domain.port.in;

import dev.pollito.spring_java.sakila.film.domain.model.Film;
import dev.pollito.spring_java.sakila.film.domain.model.FilmFilter;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface FilmUseCases {
  Film createFilm(Film film);

  Film getFilm(Integer id);

  Page<Film> getFilms(Pageable pageable);

  Page<Film> getFilms(FilmFilter filter, Pageable pageable);

  Film updateFilm(Integer id, Film film);

  void deleteFilm(Integer id);
}

File: spring_kotlin/src/main/kotlin/dev/pollito/spring_kotlin/sakila/film/domain/port/in/FilmUseCases.kt
package dev.pollito.spring_kotlin.sakila.film.domain.port.`in`

import dev.pollito.spring_kotlin.sakila.film.domain.model.Film
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface FilmUseCases {
  fun createFilm(film: Film): Film

  fun getFilm(id: Int): Film

  fun getFilms(pageable: Pageable): Page<Film>

  fun updateFilm(id: Int, film: Film): Film
}
